src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -511,14 +533,77 @@ private static bool HasDuplicateInterfaces(NamedTypeSymbol type, ConsList<Symbol
                     set.Free();
                     return false;
             }
-            
+
             // very rare case. 
             // some implemented interfaces are related
             // will have to instantiate interfaces and check
             hasRelatedInterfaces:
             return type.InterfacesNoUseSiteDiagnostics(basesBeingResolved).HasDuplicates(TypeSymbol.EqualsIgnoringDynamicAndTupleNamesComparer);
         }
 
+        internal static bool CheckTupleNamesAgainstBase(TypeSymbol type, Location location, DiagnosticBag diagnostics)
+        {
+            ArrayBuilder<DiagnosticInfo> builder = null;
+
+            bool result = CheckTupleNamesAgainstBase(type, ref builder);
+
+            if (builder != null)
+            {
+                foreach (var diagnosticInfo in builder)
+                {
+                    diagnostics.Add(new CSDiagnostic(diagnosticInfo, location));
+                }
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Check interfaces for tuple name differences with AllInterfaces from base type.
+        /// Returns false if there was a problem.
+        /// </summary>
+        internal static bool CheckTupleNamesAgainstBase(TypeSymbol type, ref ArrayBuilder<DiagnosticInfo> builder)
```

Comment:
A related question, although not from this PR: Why do methods have both `diagnosticsBuilder` and `useSiteDiagnosticsBuilder`? Wouldn't one builder be sufficient?